### ISSUE TITLE
[GradedAxes] [BUG] Iterating over GradedUnitRange yields unexpected type change

### ISSUE THREAD

**Issue by ogauthe** [ITensor/ITensors.jl#1416]
```julia
using NDTensors.Sectors: U1
using NDTensors.GradedAxes: gradedrange

g1 = gradedrange([U1(0) => 2, U1(1)=>1])
for s in g1
    println(s, " ", typeof(s))
end
```
outputs
```
NDTensors.LabelledNumbers.LabelledInteger{Int64, U1{Int64}}(1, U(1)[0]) NDTensors.LabelledNumbers.LabelledInteger{Int64, U1{Int64}}
2 Int64
3 Int64
```
with a change in iterator type beyond the first element. One obtains the same result with `s in eachindex(g1)` (I guess they are the same internally)

**Comment by mtfishman** [comment]
This will be fixed in a rewrite I will do of GradedAxes which will account for the new BlockArrays.jl v1 release.

**Comment by mtfishman** [comment]
It was easy enough to fix in #1468 without rewriting the GradedUnitRange type for now.
